testtools/tests/test_testcase.py

```diff
@@ -1967,6 +1967,39 @@ def test_other_attribute(self):
         self.assertRaises(AttributeError, getattr, orig, 'thing')
 
 
+class TestTCaseExpectThat(TestCase):
+
+    def setUp(self):
+        super(TestTCaseExpectThat, self).setUp()
+        self.passed = None
+        self.message = 'default'
+
+    def expectThat(self, matcher, matchee, message='', verbose=False):
+        passed, message = super(TestTCaseExpectThat, self).expectThat(matcher, matchee, message, verbose)
+        self.passed = passed
+        self.message = message
+        return passed, message
+
+    def test_expectthat_success(self):
+        s, m = self.expectThat(True, Equals(True))
+        assert s
+        assert s == self.passed
+        assert m is self.message is None
```

Comment:
I'd prefer to use richer comparisons than bare asserts: unlike py.test, unittest doesn't currently introspect assert statements.

e.g.

```
    def test_expectThat_success(self):
        self.assertThat(self.expectThat(True, Equals(True)), Is(None))
```

and

```
    def test_expectThat_failure(self):
        mismatch = self.expectThat(False, Equals(True))
        self.assertThat(str(mismatch), Equals(str(Equals(True).match(False))))
```

will give complete coverage I believe.